### DISCUSSION TITLE
Only load data on initial render

### DISCUSSION THREAD

**Discussion by abdulrahimiliasu** [vercel/swr#2538] [Q&A]
I would like to mutate some data e.g

```typescript
import useSWR, { mutate } from "swr";


export default function App() {
    const { data, isLoading, error , mutate } = useSWR("user/data", () =>  getData("default")));
    const [selected, setSelected] = useState(undefined);
    
  const mutateData = async () => {
    setSelected("newThing")
    const newData = await api.getData("newThing")
    mutate(newData)
 }

    return (
        <div>
            <button onClick={mutateData}></button>
        </div>
    );
}

```

How can I mutate and set a state variable (which re-renders) at the same time, because this will always set back the initial data as `data` after re-render. I'm more of looking for a pattern if there is. Thanks

**Comment by mnik01** [comment]
mutate(newData, {revalidate:false} solves your problem?

**Reply by abdulrahimiliasu** [reply]
No, because if I update a state variable after mutating, then the hook will load the initial data i.e `getData("default")` after re rendering. So it doesn't matter what I pass in `mutate`. I'm not sure if there's a way to load data only once at the beginning and only update using `mutate`. As of now it's only possible with `useEffect`.
